dsf-bpe-process-feasibility-mpc/src/main/java/org/highmed/dsf/bpe/service/DownloadResearchStudyResource.java

```diff
@@ -0,0 +1,210 @@
+	}
+
+	private List<String> getOrganizationIdentifiersOfConsortium(String consortiumIdentifier)
+	{
+		return organizationProvider
+				.getOrganizationsByConsortiumAndRole(consortiumIdentifier,
+						CODESYSTEM_HIGHMED_ORGANIZATION_ROLE_VALUE_MEDIC)
+				.flatMap(o -> o.getIdentifier().stream())
+				.filter(i -> NAMINGSYSTEM_HIGHMED_ORGANIZATION_IDENTIFIER.equals(i.getSystem())).map(i -> i.getValue())
+				.collect(Collectors.toList());
+	}
+
+	private List<String> getOrganizationIdentifiersOfResearchStudy(ResearchStudy researchStudy)
+	{
+		return researchStudy.getExtensionsByUrl(EXTENSION_HIGHMED_PARTICIPATING_MEDIC).stream()
+				.filter(e -> e.getValue() instanceof Reference).map(e -> (Reference) e.getValue())
+				.map(r -> r.getIdentifier().getValue()).collect(Collectors.toList());
+	}
+
+	private void checkConsortiumAffiliation(List<String> identifiersConsortium, List<String> identifiersResearchStudy,
+			String researchStudyId, String consortiumIdentifier)
+	{
+		List<String> identifiersWrongConsortium = new ArrayList<>(identifiersResearchStudy);
+		identifiersWrongConsortium.removeAll(identifiersConsortium);
+		if (!identifiersWrongConsortium.isEmpty())
+		{
+			logger.warn(
+					"Organizations with identifiers='{}' are part of FeasibilityMpc research study with id='{}' but do "
+							+ "not belong to the consortium with identifier='{}'",
+					identifiersWrongConsortium, researchStudyId, consortiumIdentifier);
+
+			throw new RuntimeException("Organizations with identifiers='" + identifiersWrongConsortium
+					+ "' are part of FeasibilityMpc research study with id='" + researchStudyId
+					+ "' but do not belong to the consortium with identifier='" + consortiumIdentifier + "'");
+		}
+	}
+
+	private ResearchStudy addMissingOrganizations(List<String> identifiersConsortium,
+			List<String> identifiersResearchStudy, ResearchStudy researchStudy, FhirWebserviceClient client)
+	{
+		identifiersConsortium.removeAll(identifiersResearchStudy);
+
+		if (!identifiersConsortium.isEmpty())
+		{
+			identifiersConsortium.forEach(identifier ->
+			{
+				logger.warn(
+						"Adding missing organization with identifier='{}' to FeasibilityMpc research study with id='{}'",
+						identifier, researchStudy.getId());
+
+				researchStudy.addExtension().setUrl(EXTENSION_HIGHMED_PARTICIPATING_MEDIC).setValue(
+						new Reference().setType(ResourceType.Organization.name()).setIdentifier(new Identifier()
+								.setSystem(NAMINGSYSTEM_HIGHMED_ORGANIZATION_IDENTIFIER).setValue(identifier)));
+
+			});
+
+			return update(researchStudy, client);
+		}
+		else
+			return researchStudy;
+	}
+
+	private ResearchStudy update(ResearchStudy researchStudy, FhirWebserviceClient client)
+	{
+		try
+		{
+			return client.update(researchStudy);
+		}
+		catch (Exception e)
+		{
+			logger.warn("Error while updating ResearchStudy resource: " + e.getMessage(), e);
```

Comment:
do not log the exception (stack-trace) when throwing the exception to avoid multiple stack-trace log entries.